torchvision/models/resnet.py

```diff
@@ -312,7 +314,7 @@ def _resnet(
 
 class ResNet18_Weights(WeightsEnum):
     IMAGENET1K_V1 = Weights(
-        url="https://download.pytorch.org/models/resnet18-f37072fd.pth",
+        url=model_urls.register("https://download.pytorch.org/models/resnet18-f37072fd.pth", "resnet18"),
```

Comment:
I understand this is still a draft but  I took a look anyway. I would suggest not to touch the new classes at all, because we would need to change that bit again once we remove `model_urls` for good. Instead it's probably easier and simpler to define the dict the same way as before, and instead of being a `dict` it's a custom class that only overrides `__getitem__` like you did above.

Reply:
Philip was just showcasing an idea we were discussing offline. This is not intended to be merged. I agree that the intention is to remove `model_urls` and a preferable approach wouldn't touch the existing weight classes.

Reply:
Yeah, this is a possible solution to avoid duplicating the URLs.